### ISSUE TITLE
[7.17] Upgrades from 7.17 X86 MacOS installations should automatically switch to the ARM artifacts when upgrading

### ISSUE THREAD

**Issue by cmacknz** [elastic/elastic-agent#4613]
When a 7.17.x agent installed on a MacOS system upgrades it always upgrades to the MacOS X86 version of the agent even though the ARM version has been available for some time.

Normally this is not a problem, but Elastic Defend now no longer supports running under emulation which means 7.17.x agents with Elastic Defend installed that upgrade to 8.13.0+ will have their upgrade rolled back.

Instead of requiring users to re-installed, we could have the 7.17 agent switch itself to the MacOS ARM package on upgrade automatically to avoid this.

Example of the Elastic Defend upgrade failing when attempted in an X86 MacOS installation that is running on an ARM MacOS machine:

From: https://github.com/elastic/elastic-agent/issues/4612#issuecomment-2072962092
```
[elastic_agent][info] check if endpoint service is installed
19:03:49.219
elastic_agent
[elastic_agent][info] after check if endpoint service is installed, err: exit status 135
19:03:49.219
elastic_agent
[elastic_agent][info] failed check endpoint service: exit status 135, try install
19:03:49.266
elastic_agent
[elastic_agent][error] failed accept conn info connection: accept tcp 127.0.0.1:6788: use of closed network connection
19:03:49.267
elastic_agent
[elastic_agent][error] Component state changed endpoint-default (STARTING->FAILED): failed install endpoint service: exit status 135
```

**Comment by pierrehilbert** [comment]
What do you have in mind from an implementation perspective?
Changing the 7.17.X behavior to automatically ask for the ARM binary instead of the x86 one but it will require to first upgrade to the last 7.17 patch version to have this fix?
Or do you have something else in mind?

**Comment by cmacknz** [comment]
Yes you'd have to upgrade to the latest 7.17 with our fix first for this to work. If we had a way on the web side to detect the host native architecture reliably we could just use a redirect, but I don't think there is a reliable way to do this.

**Comment by pierrehilbert** [comment]
Make sense to me, just wanted to see if you have another idea in mind.

**Comment by pierrehilbert** [comment]
> We could add CPU architecture to our User-Agent string when we make a request to download.

Yes but we will be in the same situation: we will need to upgrade to an earlier 7.17 version to get the fix in
